devtools/Makefile

```diff
@@ -22,11 +22,19 @@ DEVTOOLS_COMMON_OBJS :=				\
 
 devtools-all: $(DEVTOOLS)
 
+ifeq ($(EXPERIMENTAL_FEATURES),1)
+devtools/gen_print_wire.h: $(WIRE_GEN) wire/gen_peer_wire_csv $(wildcard wire/experimental_csv)
+	cat wire/gen_peer_wire_csv wire/experimental_csv | grep -v "cat: " | $(WIRE_GEN) --bolt --printwire --header $@ wire_type > $@
```

Comment:
`grep -v "cat: "` is weird: what cat prints cat?

This also won't work in the case where experimental_csv doesn't exist (and you don't include a rule to build it).  You worked around the dependency with $(wildcard) but I think you should actually insist it exists, and add a rule to build it.

So, commit 1 should change/remove the rule which generates wire/gen_peer_wire_csv (and the onion one) in favor of an explicit target which regenerates it (eg. `make spec-csv`?).  commit 2 should add a similar rule to make the experimental ones (eg. make spec-csv-experimental) perhaps by doing a diff or something?  You should also check in an empty wire/experimental_csv.

Then this, minus $(wildcard) logic, becomes commit 3.